src/ui/components/Pagination.tsx

```diff
@@ -5,14 +5,20 @@ export async function Pagination({
 	pageInfo,
 }: {
 	pageInfo: {
-		endCursor?: string | null;
+		baseUrl: string;
 		hasNextPage: boolean;
+		endCursor?: string | null;
+		urlSearchParams?: URLSearchParams;
```

Comment:
```suggestion
		urlSearchParams?: ReadonlyURLSearchParams;
```

Reply:
I am not sure but `ReadonlyURLSearchParams` is a internal class used by next.js e.g. in `useSearchParams` hook and it shouldn't be used here. Maybe insted I change it to `readonly urlSearchParams?: URLSearchParams;`?

Reply:
`import { ReadonlyURLSearchParams } from "next/navigation";`

Reply:
For me it's look that result type of `new URLSearchParams()` is not compatible with type `ReadonlyURLSearchParams` because I receive an type error: `Property '[INTERNAL_URLSEARCHPARAMS_INSTANCE]' is missing in type 'URLSearchParams' but required in type 'ReadonlyURLSearchParams'.`

Reply:
Hmmm yeah you're right, I didn't notice that. We should have our own Readonly… type for that. Let's ignore it for now.